internal/api/hooks.go

```diff
@@ -55,20 +74,219 @@ func (a *API) runHook(ctx context.Context, tx *storage.Connection, name string,
 	return response, nil
 }
 
-// invokeHook invokes the hook code. tx can be nil, in which case a new
+func (a *API) runHTTPHook(hookConfig conf.ExtensibilityPointConfiguration, input, output any) ([]byte, error) {
+
+	client := http.Client{
+		Timeout: DefaultHTTPHookTimeout,
+	}
+	requestURL := hookConfig.URI
+	hookLog := logrus.WithFields(logrus.Fields{
```

Comment:
Please get the log from the context so it's scoped under the request. Check the observability package.